app/controllers/StatusController.scala

```diff
@@ -251,13 +256,24 @@ class StatusController @Inject()(val landingService: LandingService,
     activities.fold(false)(ba => (ba.businessActivities -- Set(MoneyServiceBusiness, TrustAndCompanyServices)).nonEmpty)
   }
 
-  private def canCannotTradeContent(activities: Option[BusinessActivities])(implicit request: Request[AnyContent]) =
+  private def canOrCannotTradeInformation(activities: Option[BusinessActivities])(implicit request: Request[AnyContent]) =
     (hasMsb(activities), hasTcsp(activities), hasOther(activities)) match {
-      case (false, false, true) => can_cannot_trade_no_msb_or_tcsp()
-      case (true, _, false) | (_, true, false) => can_cannot_trade_msb_or_tcsp_only()
-      case (true, _, true) | (_, true, true) => can_cannot_trade()
+      case (false, false, true) => trade_information_no_msb_or_tcsp()
+      case (true, _, false) | (_, true, false) => trade_information_msb_or_tcsp_only()
+      case (true, _, true) | (_, true, true) => trade_information()
       case (_, _, _) => throw new MatchError("Could not match activities against given options.")
     }
+
+  def countUnreadNotifications(amlsRefNo: Option[String], safeId: Option[String], accountTypeId: (String, String))(implicit headerCarrier: HeaderCarrier) = {
+    val notifications = (amlsRefNo, safeId) match {
+      case (Some(ref), _) => notificationConnector.fetchAllByAmlsRegNo(ref, accountTypeId)
+      case (None, Some(id)) => notificationConnector.fetchAllBySafeId(id, accountTypeId)
+      case (None, None) => Future.successful(Seq())
+      case (_, _) => throw new MatchError("Could not match amls ref number or safe id against given conditions.")
```

Comment:
Might be less disruptive to return a zero here rather than raise an exception? An exception will cause a 'Please try again later'. If this is non recoverable it will block the user from accessing the service for an unread notification count. Maybe we could log a warning so we can trace in Kibana however; return a zero and allow them to carry on where we can't determine the count?

Reply:
Ok, thanks for the review, I'll address above matter in the morning.

Reply:
Done, default case will be empty seq, so it will return 0. Removed the MatchError and corresponding test.